ballerina/googlesheets_client.bal

```diff
@@ -0,0 +1,520 @@
+        if (dataType == "int") {
+            return int:fromString(value.toString());
+        } else if dataType == "time:Date" || dataType == "time:TimeOfDay" ||dataType == "time:Civil" || dataType == "time:Utc" {
+            
+            return self.stringToTime(value.toString(), dataType);
+        } else if (dataType == "string") {
+            return value.toString();
+        } else if (dataType == "decimal") {
+            return decimal:fromString(value.toString());
+        } else {
+            return <error>error("unsupported data format");
+        }
+    }
+
+    private isolated function generateMetadataValue(string[] keyFields, record {}|map<anydata> rowValues) returns string {
+        string metadataValue = "";
+        foreach string key in keyFields {
+            if (metadataValue != "") {
+                metadataValue += ":";
+            }
+            metadataValue += rowValues[key].toString();
+        }
+        return metadataValue;
+    }
+
+    private isolated function generateWhereClause(anydata key, map<anydata> typeMap) returns string|error {
+        string whereClause = "";
+        if (key is map<any>) {
+            foreach string primaryKey in key.keys() {
+                string? columnId = self.fieldMetadata.get(primaryKey)["columnId"];
+                if (columnId !is ()) {
+                    string keyValue = key.get(primaryKey).toString();
+                    if (whereClause != "") {
+                        whereClause += " and ";
+                    }
+                    string dataType = typeMap.get(primaryKey).toString();
+                    if (dataType == "string") {
+                        string condition = string `${columnId} = '${keyValue}'`;
+                        whereClause += condition;
+                    } else {
+                        string condition = string `${columnId} = ${keyValue}`;
+                        whereClause += condition;
+                    }
+                } else {
+                    return <error>error(string `ColumnId for the field : ${primaryKey} cannot be found.`);
+                }
+            }
+        } else {
+            string? columnId = self.fieldMetadata.get(self.keyFields[0])["columnId"];
+            if (columnId !is ()) {
+                string dataType = typeMap.get(self.keyFields[0]).toString();
+                if (dataType == "string") {
+                    whereClause = string `${columnId} = '${key.toString()}'`;
+                } else {
+                    whereClause = string `${columnId} = ${key.toString()}`;
+                }
+
+            } else {
+                return <error>error(string `ColumnId for the field : ${self.keyFields[0]} cannot be found.`);
+            }
+        }
+        return whereClause;
+
+    }
+
+    private isolated function generateColumnIds(string[] fields) returns string|error {
```

Comment:
Is this method used anywhere?

Reply:
removed in [4346c39](https://github.com/ballerina-platform/module-ballerina-persist/pull/201/commits/4346c3959a465aeb05de0872b94ea1056a715a9a)